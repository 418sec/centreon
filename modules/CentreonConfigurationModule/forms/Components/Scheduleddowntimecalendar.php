<?php
/*
 * Copyright 2005-2015 CENTREON
 * Centreon is developped by : Julien Mathis and Romain Le Merlus under
 * GPL Licence 2.0.
 *
 * This program is free software; you can redistribute it and/or modify it under
 * the terms of the GNU General Public License as published by the Free Software
 * Foundation ; either version 2 of the License.
 *
 * This program is distributed in the hope that it will be useful, but WITHOUT ANY
 * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
 * PARTICULAR PURPOSE. See the GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License along with
 * this program; if not, see <http://www.gnu.org/licenses>.
 *
 * Linking this program statically or dynamically with other modules is making a
 * combined work based on this program. Thus, the terms and conditions of the GNU
 * General Public License cover the whole combination.
 *
 * As a special exception, the copyright holders of this program give CENTREON
 * permission to link this program with independent modules to produce an executable,
 * regardless of the license terms of these independent modules, and to copy and
 * distribute the resulting executable under terms of CENTREON choice, provided that
 * CENTREON also meet, for each linked independent module, the terms  and conditions
 * of the license of that module. An independent module is a module which is not
 * derived from this program. If you modify this program, you may extend this
 * exception to your version of the program, but you are not obliged to do so. If you
 * do not wish to do so, delete this exception statement from your version.
 *
 * For more information : contact@centreon.com
 *
 */

namespace CentreonConfiguration\Forms\Components;

use Centreon\Internal\Di;
use Centreon\Internal\Form\Component\Component;

/**
 * Component for select the period
 *
 * @author Maximilien Bersoult <mbersoult@centreon.com>
 * @package CentreonConfiguration
 * @subpackage Form
 * @version 3.0.0
 */
class Scheduleddowntimecalendar extends Component
{
    /**
     * Generate and return the html for the element
     *
     * @param array $element The element to parse
     * @return array
     */
    public static function renderHtmlInput(array $element)
    {
        $tpl = Di::getDefault()->get('template');
        $weeklyDays = array(
            0 => "sunday",
            1 => "monday",
            2 => "tuesday",
            3 => "wednesday",
            4 => "thursday",
            5 => "friday",
            6 => "saturday"
        );

        $tpl->assign('element', $element);
        $tpl->assign('weeklyDays', $weeklyDays);
        $tpl->addCss('centreon.scheduled-downtime.css', 'centreon-configuration');
        $tpl->addCss('bootstrap-datetimepicker.min.css');
        $tpl->addJs('centreon.scheduled-downtime.js', 'bottom', 'centreon-configuration');
        $tpl->addJs('hogan-3.0.0.min.js');
        $tpl->addJs('bootstrap-datetimepicker.min.js');
         
        return array(
            'html' => $tpl->fetch('file:[CentreonConfigurationModule]/forms/components/scheduled_downtime_calendar.tpl'),
        );
    }
}
