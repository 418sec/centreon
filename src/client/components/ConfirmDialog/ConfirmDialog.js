import React, { Component } from 'react'
import Dialog, { DialogActions, DialogContent } from 'material-ui/Dialog'
import Button from 'material-ui/Button'
import Typography from 'material-ui/Typography'

const ConfirmDialog = ({open, content, onCancel, onConfirm}) => (
  <Dialog
    disableBackdropClick
    disableEscapeKeyDown
    maxWidth="xs"
    open={open}
  >
    <DialogContent>
      <Typography gutterBottom align="center">
        {content.body}
      </Typography>
    </DialogContent>
    <DialogActions>
      <Button onClick={onCancel} color="primary" id="cancelBtn">
        {content.cancel}
      </Button>
      <Button onClick={onConfirm} color="primary" id="confirmBtn">
        {content.confirm}
      </Button>
    </DialogActions>
  </Dialog>
)

export default ConfirmDialog