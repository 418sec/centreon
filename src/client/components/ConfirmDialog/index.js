export { default as ConfirmDialog } from './ConfirmDialogContainer';