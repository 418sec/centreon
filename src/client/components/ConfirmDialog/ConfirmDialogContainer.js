import React, { Component } from 'react'
import { connect } from 'react-redux'
import ConfirmDialog from "./ConfirmDialog"
import {
  resetAction,
  confirmAction
} from 'Redux/confirmDialog/confirmDialogActions'

class ConfirmDialogContainer extends Component {

  handleCancel = () => {
    const { resetAction } = this.props
    resetAction()
  }

  handleConfirm = () => {
    const { confirmAction, slug } = this.props
    confirmAction(slug)
  }

  render () {
    const { open, content } = this.props

    return (
      <ConfirmDialog
        open={open}
        content={content}
        onCancel={this.handleCancel}
        onConfirm={this.handleConfirm}
      />
    )
  }
}

const mapStateToProps = (store) => {
  return {
    slug: store.confirmDialogReducer.slug,
    open: store.confirmDialogReducer.open,
    content: store.confirmDialogReducer.content,
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    resetAction: () => {
      return dispatch(resetAction())
    },
    confirmAction: (slug) => {
      return dispatch(confirmAction(slug))
    },
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(ConfirmDialogContainer)