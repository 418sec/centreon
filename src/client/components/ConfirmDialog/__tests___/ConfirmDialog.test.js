/* eslint-env jest */
import React from 'react'
import Enzyme from 'enzyme'
import { mount, shallow, render } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import { ConfirmDialog } from 'Components/ConfirmDialog'
import configureStore from 'redux-mock-store'

Enzyme.configure({ adapter: new Adapter() })

describe('ConfirmDialog', () => {
  const mockStore = configureStore()
  const store = mockStore({
    confirmDialogReducer: {
      slug: 'centreon',
      allowAction: false,
      open: true,
      content: {
        body: 'message',
        cancel: 'cancel',
        confirm: 'confirm',
      },
    }
  })

  it('Opens dialog correctly', () => {

    const ConfirmDialogContainer = shallow(
      <ConfirmDialog store={store} slug="centreon"/>
    )

    expect(toJson(ConfirmDialogContainer)).toMatchSnapshot()
  })

  it('Cancel button works correctly', () => {
    const ConfirmDialogContainer = mount(
      <ConfirmDialog store={store} slug="centreon"/>,
    )

    ConfirmDialogContainer
      .find('button#cancelBtn')
      .simulate('click')

    const actions = store.getActions()
    expect(actions[0].type).toEqual('RESET_ACTION')
  })

  it('Confirm button works correctly', () => {

    const ConfirmDialogContainer = mount(
      <ConfirmDialog store={store} slug="centreon"/>,
    )

    ConfirmDialogContainer
      .find('button#confirmBtn')
      .simulate('click')

    const actions = store.getActions()
    expect(actions[1].type).toEqual('CONFIRM_ACTION')
  })
})