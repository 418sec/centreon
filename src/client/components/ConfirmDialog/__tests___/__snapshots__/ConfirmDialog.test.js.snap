// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmDialog Opens dialog correctly 1`] = `
<ConfirmDialogContainer
  confirmAction={[Function]}
  content={
    Object {
      "body": "message",
      "cancel": "cancel",
      "confirm": "confirm",
    }
  }
  open={true}
  resetAction={[Function]}
  slug="centreon"
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
